R/epidist_utils.R

```diff
@@ -0,0 +1,703 @@
+#' @param dispersion_ci_limits A numeric vector of length two of the confidence
+#' interval around the dispersion
+#' @param dispersion_ci A numeric specifying the confidence interval
+#' width of the dispersion parameter
+#' @param lower_range The lower range of the data, used to infer the parameters
+#' of the distribution when not provided
+#' @param upper_range The upper range of the data, used to infer the parameters
+#' of the distribution when not provided
+#' @param q_025 The 2.5th quantile of the probability distribution
+#' @param q_05 The 5th quantile of the probability distribution
+#' @param q_25 The 25th quantile of the probability distribution
+#' @param q_75 The 75th quantile of the probability distribution
+#' @param q_875 The 87.5th quantile of the probability distribution
+#' @param q_95 The 95th quantile of the probability distribution
+#' @param q_975 The 97.5th quantile of the probability distribution
+#'
+#' @return A nested list of summary statistics. The highest level are
+#' - `$central_tendency_spread`
+#' - `$quantiles`
+#' - `$range`
+#' - `$dispersion`
+#' @export
+#'
+#' @examples
+#' # mean and standard deviation
+#' create_epidist_summary_stats(mean = 5, sd = 2)
+#'
+#' # mean and standard deviation with uncertainty
+#' create_epidist_summary_stats(
+#'   mean = 4,
+#'   mean_ci_limits = c(2.1, 5.7),
+#'   mean_ci = 95,
+#'   sd = 0.7,
+#'   sd_ci_limits = c(0.3, 1.1),
+#'   sd_ci = 95
+#' )
+#'
+#' # median and range
+#' create_epidist_summary_stats(
+#'   median = 5,
+#'   lower_range = 1,
+#'   upper_range = 13
+#' )
+create_epidist_summary_stats <- function(mean = NA_real_,
+                                         mean_ci_limits = c(NA_real_, NA_real_),
+                                         mean_ci = NA_real_,
+                                         sd = NA_real_,
+                                         sd_ci_limits = c(NA_real_, NA_real_),
+                                         sd_ci = NA_real_,
+                                         median = NA_real_,
+                                         median_ci_limits = c(NA_real_, NA_real_),
+                                         median_ci = NA_real_,
+                                         dispersion = NA_real_,
+                                         dispersion_ci_limits = c(NA_real_, NA_real_),
+                                         dispersion_ci = NA_real_,
+                                         lower_range = NA_real_,
+                                         upper_range = NA_real_,
+                                         q_025 = NA_real_,
+                                         q_05 = NA_real_,
+                                         q_25 = NA_real_,
+                                         q_75 = NA_real_,
+                                         q_875 = NA_real_,
+                                         q_95 = NA_real_,
+                                         q_975 = NA_real_) {
```

Comment:
These quantiles were chosen as they are the most commonly reported in the literature. For now I will keep the database the same with these quantiles, but given Seb's suggestion in this PR I will change it so that quantiles are provided by the user instead of being predefined.